strawberryfields/backends/states.py

```diff
@@ -1405,3 +1415,412 @@ def all_fock_probs(self, **kwargs):
         mu = self._mu
         cov = self._cov
         return twq.probabilities(mu, cov, cutoff, hbar=self._hbar)
+
+
+class BaseBosonicState(BaseState):
+    r"""Class for the representation of quantum states as superpositions of
+    Gaussian states.
+    """
+    # pylint: disable=too-many-public-methods
+
+    def __init__(self, state_data, num_modes, num_weights, mode_names=None):
+        super().__init__(num_modes, mode_names)
+
+        self._data = state_data
+        self.num_weights = num_weights
+        # vector of means and covariance matrix, using frontend x,p scaling
+        self._mus = self._data[0] * np.sqrt(self._hbar / 2)
+        self._covs = self._data[1] * (self._hbar / 2)
+        self._weights = self._data[2]
+
+        self._basis = "bosonic"
+        self._str = "<BosonicState: num_modes={}, num_weights={}, pure={}, hbar={}>".format(
+            self.num_modes, self.num_weights, self._pure, self._hbar
+        )
+
+    def __eq__(self, other):
+        """Equality operator for BaseBosonicState.
+
+        Returns True if other BaseBosonicState is close to self.
+        This is done by comparing the weights, means vectors and covs matrices.
+        If both are within the EQ_TOLERANCE, True is returned.
+
+        Args:
+            other (BaseGaussianState): BaseGaussianState to compare against.
+        """
+        # pylint: disable=protected-access
+        # TODO: check equality for two different representations of the same state.
```

Comment:
Fundamentally, two physical states are the same if their Wigner functions are the same. However, you can have different mathematical expressions (representations) that yield the same Wigner function. This just checks that the mathematical expressions are the same, not that the states are the same, since to check the states is harder to do.

Reply:
When you have just one weight, as is the case for Gaussian states, the state is uniquely defined by the mean and covariance, so it is sufficient to just check that those are equal for two Gaussian states.